Core/Src/segment.c

```diff
@@ -44,6 +44,12 @@ void SegmentInterface::retrieveSegmentData(ChipData_t databuf[NUM_CHIPS])
     memcpy(previous_data, segment_data, sizeof(ChipData_t)*NUM_CHIPS);
 
     segment_data = nullptr;
+
+    if (voltage_error) {
```

Comment:
so this doesnt work perfectly. What if there is a therm error and a voltage error? you will miss the therm error. Our faults work in a way that you can use |= operator. This allows u to have a value that shows both faults. ie if voltage error is represented by bit 6, and therm error is bit 7, you can do like Val |= voltage error and do val |= therm error and the value wqould be like 1100 0000 if they both faulted. This is how u should return these values

Reply:
We talked and established that the both errors are internal cell communication faults, so a therm and voltage read error are both covered by the one error code.